app/controllers/application.js

```diff
@@ -16,6 +16,24 @@ export default Controller.extend(queryParams.Mixin, {
   ajax: service(),
   store: service(),
 
+  init() {
+    this._super(...arguments);
+    // create the one-signal-player when user subscribes
+    window.OneSignal.push(() => {
+      window.OneSignal.on('subscriptionChange', (isSubscribed) => {
```

Comment:
Move to a private function in the route and call it from the authentication hook there.

May need to check that the application route isn't destroyed (I don't think it is, but don't know off top of my head)